blocks/api/registration.js

```diff
@@ -55,6 +57,15 @@ const POST_FORMAT_BLOCK_MAP = {
 	video: 'core/video',
 };
 
+const isEditPropValid = ( settings ) => {
```

Comment:
Two things:

- This seems like an enhancement we should support separate from `forwardRef` integration into `createHigherOrderComponent`, and could be its own pull request.
- This is definitely a piercing of the "element" abstraction. It seems like a function which should be offered by `@wordpress/element`. On glance, it appears similar to but not quite the same as what's already provided through `isValidElement`. Digging further, I found there is the [official `react-is` module](https://www.npmjs.com/package/react-is) which includes an `isValidElementType` method. We'd have to decide how we want to fit this into our abstraction. Honestly, I'm not sure why there's a `React.isValidElement` but not also the `isValidElementType` bundled; they seem appropriate together, and with that in mind I wouldn't mind if our package exported `isValidElementType` (though leveraging `react-is` underneath).